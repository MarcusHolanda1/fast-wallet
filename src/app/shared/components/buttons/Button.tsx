import {
  TouchableOpacity,
  Text,
  DimensionValue,
  StyleSheet,
  ActivityIndicator
} from 'react-native';
import { theme } from '@shared/theme/theme';

interface ButtonProps {
  title: string;
  onPress: () => void;
  disabled?: boolean;
  backgroundColor?: string;
  textColor?: string;
  width?: DimensionValue;
  isLoading?: boolean;
}

export default function Button({
  title,
  onPress,
  disabled,
  backgroundColor,
  textColor = '#fff',
  width = '100%',
  isLoading = false
}: ButtonProps) {
  return (
    <TouchableOpacity
      style={[
        styles.container,
        { backgroundColor, width },
        disabled && styles.disabled
      ]}
      onPress={onPress}
      disabled={disabled}
    >
      {isLoading ? (
        <ActivityIndicator size="small" color={textColor} />
      ) : (
        <Text style={[styles.text, { color: textColor }]}>{title}</Text>
      )}
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  container: {
    paddingVertical: 10,
    borderRadius: 12,
    alignItems: 'center',
    justifyContent: 'center'
  },
  text: {
    fontSize: theme.typography.h5.fontSize,
    fontWeight: '600'
  },
  disabled: {
    opacity: 0.5
  }
});
