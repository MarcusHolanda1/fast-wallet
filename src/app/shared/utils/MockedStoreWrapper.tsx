import { Provider } from 'react-redux';
import { configureStore } from '@reduxjs/toolkit';

const createTestStore = () => {
  return configureStore({
    reducer: {
      test: (state: Record<string, unknown> = {}) => state
    }
  });
};

export const TestWrapper = ({ children }: { children: React.ReactNode }) => {
  const store = createTestStore();
  return <Provider store={store}>{children}</Provider>;
};
