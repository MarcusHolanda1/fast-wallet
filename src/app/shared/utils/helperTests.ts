import { fireEvent, RenderResult } from '@testing-library/react-native';

export const fillAndBlur = (
  sut: RenderResult,
  testId: string,
  value: string
) => {
  fireEvent.changeText(sut.getByTestId(testId), value);
  fireEvent(sut.getByTestId(testId), 'blur');
};

export const expectFormattedCardNumber = (cardNumber: string) => {
  return '•••• •••• •••• ' + cardNumber.slice(-4);
};
