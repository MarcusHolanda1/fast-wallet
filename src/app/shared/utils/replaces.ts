export const replaceWithTextOnly = (text: string) => {
  return text.replace(/[^a-zA-ZÃ€-Ã¿\s]/g, '');
};

export const formatCardNumber = (cardNumber: string) => {
  return cardNumber.slice(-4);
};
