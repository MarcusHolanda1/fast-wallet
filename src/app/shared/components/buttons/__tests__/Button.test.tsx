import { render, RenderResult } from '@testing-library/react-native';
import { TestWrapper } from '@app/shared/utils/MockedStoreWrapper';

import Button from '../Button';

const backgroundColor = '#000';
const textColor = '#fff';

const makeSut = (isLoading: boolean = false) => {
  return render(
    <TestWrapper>
      <Button
        onPress={jest.fn}
        title="Test Button"
        backgroundColor={backgroundColor}
        textColor={textColor}
        testID="button-component"
        disabled={false}
        isLoading={isLoading}
        width={200}
      />
    </TestWrapper>
  );
};

describe('Button test', () => {
  let sut: RenderResult;

  beforeEach(() => {
    jest.clearAllMocks();
    sut = makeSut();
  });

  test('should render button correctly', () => {
    expect(sut.getByTestId('button-component')).toBeTruthy();
    expect(sut.getByText('Test Button')).toBeTruthy();
  });

  test('should render loading indicator when isLoading is true', () => {
    sut = makeSut(true);
    expect(sut.getByTestId('button-component')).toBeTruthy();
    expect(sut.queryByText('Test Button')).toBeNull();
    expect(sut.getByTestId('activity-indicator')).toBeTruthy();
  });

  test('should match snapshot', () => {
    expect(sut.toJSON()).toMatchSnapshot();
  });
});
