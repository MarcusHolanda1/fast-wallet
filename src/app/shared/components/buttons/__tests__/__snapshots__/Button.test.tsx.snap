// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Credit card should match snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#000",
      "borderRadius": 12,
      "height": 55,
      "justifyContent": "center",
      "opacity": 1,
      "paddingVertical": 10,
      "width": 200,
    }
  }
  testID="button-component"
>
  <Text
    style={
      [
        {
          "fontSize": 18,
          "fontWeight": "400",
          "lineHeight": 20,
        },
        {
          "color": "#fff",
        },
      ]
    }
  >
    Test Button
  </Text>
</View>
`;
